common/ayon_common/utils.py

```diff
@@ -414,6 +416,49 @@ def get_downloads_dir():
     return path
 
 
+class ZipFileLongPaths(zipfile.ZipFile):
+    """Allows longer paths in zip files.
+
+    Regular DOS paths are limited to MAX_PATH (260) characters, including
+    the string's terminating NUL character.
+    That limit can be exceeded by using an extended-length path that
+    starts with the '\\?\' prefix.
+    """
+    _is_windows = platform.system().lower() == "windows"
+
+    def _extract_member(self, member, tpath, pwd):
+        if self._is_windows:
+            tpath = os.path.abspath(tpath)
+            if tpath.startswith("\\\\"):
+                tpath = "\\\\?\\UNC\\" + tpath[2:]
+            else:
+                tpath = "\\\\?\\" + tpath
+
+        return super(ZipFileLongPaths, self)._extract_member(
+            member, tpath, pwd
+        )
+
+
+def get_archive_ext_and_type(archive_file):
+    """Get archive extension and type.
+
+    Args:
+        archive_file (str): Path to archive file.
+
+    Returns:
+        Tuple[str, str]: Archive extension and type.
+    """
+
+    ext = os.path.splitext(archive_file)[1].lower()
+    if ext == ".zip":
+        return ext, "zip"
+
+    if ext in {".tar", ".tgz", ".tar.gz", ".tar.xz", ".tar.bz2"}:
```

Comment:
This will fail with `.tar.gz` (or any of the "compounded ones"):
```
>>> archive_file = "AYON-0.0.2-linux.tar.gz"
>>> ext = os.path.splitext(archive_file)[1].lower()
>>> ext
'.gz'
>>> ext in {".tar", ".tgz", ".tar.gz", ".tar.xz", ".tar.bz2"}
False
```

I don't think there's any way around it without using the `endswith` , there's `pathlibs`:
```
>>> from pathlib import Path
>>> p = Path(archive_file)
>>> p.suffixes
['.0', '.2-linux', '.tar', '.gz']
```
but it would piggyback on assumptions.


So I think the way about this would be:
```py
if archive_file.lower().endswith(".zip"):
    return ".zip", "zip"
elif archive_file.lower().endswith((".tar", ".tar.gz", ".tar.xz", ".tar.bz2")):
    return  os.path.splitext(archive_file)[1].lower(), "tar"
elif archive_file.lower().endswith(".tgz"):
    return ".tgz", "tar"
```

Reply:
Aaah, good point, should be fixed.